lib/puppet/provider/ios_acl_entry/ios.rb

```diff
@@ -0,0 +1,453 @@
+    end
+    unless instance[:ack].nil?
+      instance[:ack] = 'ack'
+    end
+    unless instance[:dscp].nil?
+      instance[:dscp] = "dscp #{instance[:dscp]}"
+    end
+    unless instance[:fin].nil?
+      instance[:fin] = 'fin'
+    end
+    unless instance[:fragments].nil?
+      instance[:fragments] = 'fragments'
+    end
+    unless instance[:log].nil?
+      instance[:log] = 'log'
+    end
+    unless instance[:log_input].nil?
+      instance[:log_input] = 'log_input'
+    end
+    unless instance[:match_all].nil?
+      instance[:match_all] = "match-all #{instance[:match_all].join(' ')}"
+    end
+    unless instance[:match_any].nil?
+      instance[:match_any] = "match-any #{instance[:match_any].join(' ')}"
+    end
+    unless instance[:option].nil?
+      instance[:option] = "option #{instance[:option]}"
+    end
+    unless instance[:precedence].nil?
+      instance[:precedence] = "precedence #{instance[:precedence]}"
+    end
+    unless instance[:psh].nil?
+      instance[:psh] = 'psh'
+    end
+    unless instance[:reflect].nil?
+      instance[:reflect] = "reflect #{instance[:reflect]}"
+    end
+    unless instance[:reflect_timeout].nil?
+      instance[:reflect_timeout] = "timeout #{instance[:reflect_timeout]}"
+    end
+    unless instance[:rst].nil?
+      instance[:rst] = 'rst'
+    end
+    unless instance[:syn].nil?
+      instance[:syn] = 'syn'
+    end
+    unless instance[:time_range].nil?
+      instance[:time_range] = "time-range #{instance[:time_range]}"
+    end
+    unless instance[:tos].nil?
+      instance[:tos] = "tos #{instance[:tos]}"
+    end
+    unless instance[:urg].nil?
+      instance[:urg] = 'urg'
+    end
+
+    command = PuppetX::CiscoIOS::Utility.set_values(instance, commands_hash)
+    if instance[:ensure].to_s == 'absent'
+      command = 'no ' + command
+    end
+    commands << command
+    commands
+  end
+
+  def commands_hash
+    Puppet::Provider::IosAclEntry::IosAclEntry.commands_hash
+  end
+
+  def get(context)
+    output = context.device.run_command_enable_mode(PuppetX::CiscoIOS::Utility.get_values(commands_hash))
+    return [] if output.nil?
+    Puppet::Provider::IosAclEntry::IosAclEntry.instances_from_cli(output)
+  end
+
+  def set(context, changes)
+    changes.each do |name, change|
+      # What type of ACL are we using
+      acl_type = Puppet::Provider::IosAclEntry::IosAclEntry.type_of_access_list(context.device.run_command_enable_mode("show ip access-lists #{change[:should][:access_list]}"))
+
+      is = change.key?(:is) ? change[:is] : (get(context) || []).find { |key| key[:name] == name }
+      should = change[:should]
+      if should[:ensure].to_s == 'absent'
+        if Puppet::Provider::IosAclEntry::IosAclEntry.name_of_access_list(context.device.run_command_enable_mode("show ip access-lists #{change[:should][:access_list]}")).nil?
+          raise "ios_acl_entry delete requires parent ios_access_list #{change[:should][:access_list]} to be already present"
```

Comment:
We may also want to look at using an autorequire for acl_entry -> access_list

Reply:
Can we discuss this - the acl_entry would also need to know the *type* of the ACL (eg. extended, standard, reflexive), is this something we want to add to the acl_entry type?

Reply:
Done